src/api/helpers/ProposalValidator.ts

```diff
@@ -0,0 +1,41 @@
+// This checks, whether the payload of a proposal is in fact the same as the submitted data
+
+import MatriculationData from "../api_models/matriculation_management/MatriculationData";
+import SubjectMatriculation from "../api_models/matriculation_management/SubjectMatriculation";
+import Proposal from "../api_models/transactions/Proposal";
+import { isEqual } from "lodash";
+
+export function validateMatriculationProposal(enrollmentId: string, matriculation: SubjectMatriculation[], proposal: Proposal): boolean {
+    const txName = proposal.payload.input.input.args[0];
+
+    if (txName !== "UC4.Approval:approveTransaction") return false;
```

Comment:
should we maybe consider moving this into the `.env` file thus make it configurable in case something changes on hyperledger site?